Sources/Bot/Services/MergeService.swift

```diff
@@ -17,13 +17,33 @@ public final class MergeService {
     private let statusChecksCompletion: Signal<StatusEvent, NoError>
     internal let statusChecksCompletionObserver: Signal<StatusEvent, NoError>.Observer
 
+    /// Instantiate a MergeService for a given target branch.
+    ///
+    /// A MergeService is a state machine responsible for handling a queue of Pull Requests,
+    /// merging them in the right order to avoid multiple PRs from conflicting or affecting one another once merged.
+    ///
+    /// An instance of a MergeService is responsible for all the Pull Requests targeting the branch it is responsible for.
+    /// There can be multiple MergeServices instances in parallel, each handling the PRs for a different target branch,
+    /// (since it's still safe to merge Pull Requests targeting different branches in parallel).
+    /// Those different MergeServices per target branch are typically managed by a `DispatchService`
+    ///
+    /// - Parameter targetBranch: The target branch this MergeService is responsible for
```

Comment:
[nit] there is a format to group all parameters without prefixing each line (might be a bit different that the suggestion in reality)
 
```suggestion
    /// Parameters:
    /// - targetBranch: The target branch this MergeService is responsible for
```

Reply:
I created this doc-comment using Xcode auto-generation feature (Cmd-Alt-/ or something) so I just went with what it suggested 😄 
But yeah will try the alternate presentation, would make it nicer.